--- a/build.rs
+++ b/build.rs
@@ -37,6 +37,10 @@
     let filename: &str = "bochscpu-build-ubuntu-latest-x64.zip";
     #[cfg(all(target_arch = "aarch64", target_os = "linux"))]
     let filename: &str = "bochscpu-build-ubuntu-24.04-arm-arm64.zip";
+    #[cfg(all(target_arch = "x86_64", target_os = "macos"))]
+    let filename: &str = "bochscpu-build-macos-latest-x64.zip";
+    #[cfg(all(target_arch = "aarch64", target_os = "macos"))]
+    let filename: &str = "bochscpu-build-macos-latest-arm64.zip";

     let asset = js["assets"]
         .members()
@@ -57,7 +61,7 @@
         "Release"
     };

-    #[cfg(target_os = "linux")]
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
     let tempfile = std::path::PathBuf::from(format!(
         "{}/bochscpu-build-{}.zip",
         std::env::var("TEMP").unwrap_or("/tmp".to_string()),
